widget/nsIWidget.h

```diff
@@ -2087,8 +2087,18 @@ class nsIWidget : public nsISupports {
     virtual void SetDrawMouse(bool aDrawMouse)
     { mDrawMouse = aDrawMouse; }
 
+    virtual void SetDrawVirtualMouse(bool aDrawVirtualMouse)
+    { mDrawVirtualMouse = aDrawVirtualMouse; }
+
     virtual bool GetDrawMouse()
-    { return mDrawMouse; }
+    { return mDrawMouse || mDrawVirtualMouse; }
+
+    /**
+     * For remote control draw mouse cursor in nsContentUtils.
+     * widget/gonk/nsWindow.cpp has real implementation.
+     * Otherwise, use empty implementation.
+     */
+    virtual void ForceComposition() { };
```

Comment:
I'm thinking to scope these interfaces with #ifdef MOZ_WIDGET_GONK.
Or move them into another pure interface class for gonk::nsWindow.
Or move these interfaces and member variable into small singleton that only lives in gonk platform.

What do you think ?